### PR TITLE
BAEL-3603 Add test examples for Java IO versus NIO article

### PR DESCRIPTION
**Pull request by rojyates**

### CODE REVIEW

File: core-java-modules/core-java-io-2/src/test/java/com/baeldung/blockingnonblocking/NonBlockingClientUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    @Before
+    public void setup() {
+        stubFor(get(urlEqualTo(REQUESTED_RESOURCE)).willReturn(aResponse()
+          .withStatus(200)
+          .withBody("{ \"response\" : \"It worked!\" }")));
+    }
+
+    @Test
+    public void givenJavaNIOSocketChannel_whenReadingAndWriting_thenUseBuffers() throws IOException {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-io-2/src/test/java/com/baeldung/blockingnonblocking/NonBlockingClientUnitTest.java]:
```suggestion
    public void givenJavaNIOSocketChannel_whenReadingAndWritingWithBuffers_thenSuccess() throws IOException {
```

File: core-java-modules/core-java-io-2/src/test/java/com/baeldung/blockingnonblocking/NonBlockingClientUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        socketChannel.close();
+
+        // then we read and saved our data
+        assertTrue(ourStore
+          .toString()
+          .contains("It worked!"));
+    }
+
+    @Test
+    public void givenJavaNIO_whenReadingAndWriting_thenSmallBuffers() throws IOException {
```
Comment (KevinGilmore) [inline core-java-modules/core-java-io-2/src/test/java/com/baeldung/blockingnonblocking/NonBlockingClientUnitTest.java]:
```suggestion
    public void givenJavaNIO_whenReadingAndWritingWithSmallBuffers_thenSuccess() throws IOException {
```

File: core-java-modules/core-java-io-2/src/test/java/com/baeldung/blockingnonblocking/NonBlockingClientUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        // given a NIO SocketChannel and a charset
+        InetSocketAddress address = new InetSocketAddress("localhost", wireMockRule.port());
+        SocketChannel socket = SocketChannel.open(address);
```
Comment (KevinGilmore) [inline core-java-modules/core-java-io-2/src/test/java/com/baeldung/blockingnonblocking/NonBlockingClientUnitTest.java]: Recommend naming this variable socketChannel to be consistent with the previous test

File: core-java-modules/core-java-io-2/src/test/java/com/baeldung/blockingnonblocking/NonBlockingClientUnitTest.java (hunk 1 above)
Comment (rojyates) [reply core-java-modules/core-java-io-2/src/test/java/com/baeldung/blockingnonblocking/NonBlockingClientUnitTest.java]: Resolved this in the latest commit

File: core-java-modules/core-java-io-2/src/test/java/com/baeldung/blockingnonblocking/NonBlockingClientUnitTest.java (hunk 2 above)
Comment (rojyates) [reply core-java-modules/core-java-io-2/src/test/java/com/baeldung/blockingnonblocking/NonBlockingClientUnitTest.java]: Resolved this in the latest commit

File: core-java-modules/core-java-io-2/src/test/java/com/baeldung/blockingnonblocking/NonBlockingClientUnitTest.java (hunk 3 above)
Comment (rojyates) [reply core-java-modules/core-java-io-2/src/test/java/com/baeldung/blockingnonblocking/NonBlockingClientUnitTest.java]: Resolved this in the latest commit

**Pull request merged**